frontend/src/components/pages/CampCreation/ScheduleSessions/AddSessionsForm.tsx

```diff
@@ -0,0 +1,198 @@
+  };
+
+  const [sessionDaysHasError, setSessionDaysHasError] = React.useState(false);
+
+  // get dates of the selected week days within a week of the start date
+  const getSessionDates = (
+    sessionStartDate: Date,
+    selectedWeekDayValues: boolean[],
+  ): Date[] => {
+    const dates: Date[] = [];
+
+    let currDay = sessionStartDate.getDay();
+    for (
+      let daysAfterStartDate = 0;
+      daysAfterStartDate < 7;
+      daysAfterStartDate += 1
+    ) {
+      // only add days that the user selected
+      // e.g. only add Mondays - Fridays dates, don't add weekends
+      if (selectedWeekDayValues[currDay]) {
+        const newDate = new Date(sessionStartDate.getTime());
+        newDate.setDate(newDate.getDate() + daysAfterStartDate);
+        dates.push(newDate);
+      }
+      currDay = (currDay + 1) % 7;
+    }
+    return dates;
+  };
+
+  const onAddSesssionsToCamp = (e: any) => {
+    if (noSessionDaysSelected()) {
+      setSessionDaysHasError(true);
+    } else {
+      const updatedSessions: CreateCampSession[] = Object.assign(
+        [],
+        scheduledSessions,
+      );
+      const selectedWeekDayValues = Array.from(selectedWeekDays.values());
+
+      // create a new session a week from the start date for each successive session
+      // 0 successive sessions means create just one session from the start date
+      // 1 success session means create one session from the start date and another session a week after
+      for (let i = 0; i <= successiveSessions; i += 1) {
+        const sessionStartDate = new Date(startDate.getTime());
+        sessionStartDate.setDate(sessionStartDate.getDate() + i * 7);
+
+        const newCampSession: CreateCampSession = {
+          startDate: sessionStartDate,
+          endDate: new Date(),
+          dates: [],
+          selectedWeekDays,
+        };
+
+        const dates: Date[] = getSessionDates(
+          sessionStartDate,
+          selectedWeekDayValues,
+        );
+        newCampSession.dates = dates;
+        // eslint-disable-next-line prefer-destructuring
+        newCampSession.startDate = dates[0];
+        newCampSession.endDate = dates[dates.length - 1];
+        updatedSessions.push(newCampSession);
+      }
+
+      // scheduled sessions need to be sorted by start date
+      updatedSessions.sort(
+        (a, b) => a.startDate.getTime() - b.startDate.getTime(),
+      );
+      setScheduledSessions(updatedSessions);
+
+      // reset form states
+      setStartDate(new Date());
+      setSuccessiveSessions(0);
+      setSelectedWeekDays(new Map(emptyWeekDays));
+
+      // hide form and show scheduled sessions list
+      setShowAddSessions(false);
+    }
+    e.preventDefault();
+  };
+
+  return (
+    <Box paddingX="64px" paddingY="80px">
+      <Text textStyle="displayLarge" marginBottom={10}>
+        Add Camp Session(s)
+      </Text>
+      <form onSubmit={onAddSesssionsToCamp}>
```

Comment:
kept the form tag because without it the form controls would stop validating - @dhmi19 if there is a Chakra UI form component could you send me a link? cuz I can't find it and FormControl doesn't have the same behaviour as form